TicketsDemo.Domain/DefaultImplementations/DefaultPriceCalculationStrategy.cs

```diff
@@ -14,30 +16,32 @@ public class DefaultPriceCalculationStrategy : IPriceCalculationStrategy
         private IRunRepository _runRepository;
         private ITrainRepository _trainRepository;
 
-        public DefaultPriceCalculationStrategy(IRunRepository runRepository, ITrainRepository trainRepository) {
+        public DefaultPriceCalculationStrategy(IRunRepository runRepository, ITrainRepository trainRepository)
+        {
             _runRepository = runRepository;
             _trainRepository = trainRepository;
         }
-
-        public List<PriceComponent> CalculatePrice(PlaceInRun placeInRun)
+       
+        public List<PriceComponent> CalculatePrice(PriceCalculationParametersDTO teaCoffeeBedParametrs)
         {
-            var components = new List<PriceComponent>();
+            var Decorator = new TeaCoffeeBedDecorator(new TeaCoffeeBedPriceCalcStrategy());
```

Comment:
This is the violation of DependencyInversion principle.
Because you reference implementations from another implementations.